### PR TITLE
Use head span for `rustc_on_unimplemented`'s `enclosing_scope` attr

### PR DESCRIPTION
**Pull request by compiler-errors**
This may make #101281 slightly easier to understand

### CODE REVIEW

**Comment by compiler-errors** [comment]
I am also happy to rename `enclosing_scope` now that it's really not "enclosing" -- it's more like `parent_label` now.

File: compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs [hunk 1]
```diff
@@ -526,7 +526,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                                 });
 
                             let enclosing_scope_span =
-                                tcx.hir().span_with_body(tcx.hir().local_def_id_to_hir_id(body));
+                                tcx.hir().span(tcx.hir().local_def_id_to_hir_id(body));
```
Comment (cjgillot) [inline compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs:529]:
```suggestion
                                tcx.def_span(body);
```

**Comment by oli-obk** [comment]
r=me with def span used and the proposed variable rename applied

**Pull request merged**